modules/builds-configuration-file.adoc

```diff
@@ -0,0 +1,92 @@
+// Module included in the following assemblies:
+//
+// * builds/build-configuration.adoc
+
+[id="builds-configuration-file_{context}"]
+= Configuring build settings
+
+You can configure build settings by editing the
+`build.config.openshift.io/cluster` resource.
+
+.Procedure
+
+* Edit the `build.config.openshift.io/cluster` resource:
++
+----
+$ oc edit build.config.openshift.io/cluster
+----
++
+The following is an example `build.config.openshift.io/cluster` resource:
++
+[source,yaml]
+----
+apiVersion: config.openshift.io/v1
+kind: Build<1>
+metadata:
+  annotations:
+    release.openshift.io/create-only: "true"
+  creationTimestamp: "2019-05-17T13:44:26Z"
+  generation: 2
+  name: cluster
+  resourceVersion: "107233"
+  selfLink: /apis/config.openshift.io/v1/builds/cluster
+  uid: e2e9cc14-78a9-11e9-b92b-06d6c7da38dc
+spec:
+  additionalTrustedCA:<2>
+    name: myconfigmap
+  buildDefaults:<3>
+    defaultProxy:<4>
+      httpProxy: http://proxy.com
+      httpsProxy: https://proxy.com
+      noProxy: internal.com
+    env:<5>
+    - name: envkey
+      value: envvalue
+    gitProxy:<6>
+      httpProxy: http://gitproxy.com
+      httpsProxy: https://gitproxy.com
+      noProxy: internalgit.com
+    imageLabels:<7>
+    - name: labelkey
+      value: labelvalue
+    resources:<8>
+      limits:
+        cpu: 100m
+        memory: 50Mi
+      requests:
+        cpu: 10m
+        memory: 10Mi
+  buildOverrides:<9>
+    imageLabels:<10>
+    - name: labelkey
+      value: labelvalue
+    nodeSelector:<11>
+      selectorkey: selectorvalue
+    tolerations:<12>
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/builds
+operator: Exists
+----
+<1> `Build`: Holds cluster-wide information on how to handle builds. The
+canonical name is `cluster`.
+<2> `additionalTrustedCA`: A reference to a ConfigMap containing additional CAs that
+should be trusted for image pushes and pulls during builds. The namespace for
+this ConfigMap is `openshift-config`.
+<3> `buildDefaults`: Controls the default information for builds.
+<4> `defaultProxy`: Contains the default proxy settings for all build operations,
+including image pull or push and source download.
+<5> `env`: A set of default environment variables that are applied to the build if
+the specified variables do not exist on the build.
+<6> `gitProxy`: Contains the proxy settings for Git operations only. If set, this
+overrides any Proxy settings for all Git commands, such as `git clone`.
```

Comment:
i guess this also overrides the defaultProxy, right @adambkaplan ?

Reply:
@bparees no- this only sets the proxy for the git clone init container.

Reply:
@adambkaplan I phrased it poorly but what i meant is "if you have a defaultProxy set and then you also set this, this value takes precedence(overrides) for git clone operations".

(and if you only have defaultProxy set, it is used for git clone).

it might be worth stating that clearly so people understand the relationship between defaultProxy and gitProxy (assuming i'm remembering it correctly, i think we went through some iterations on it)